src/main/java/org/javawebstack/abstractdata/AbstractArray.java

```diff
@@ -343,4 +343,38 @@ public AbstractElement clone() {
         return new AbstractArrayCollector<>(e -> e);
     }
 
+    public boolean equals(Object obj, boolean strict) {
+        if (obj == null)
+            return false;
+        if (!(obj instanceof AbstractArray))
```

Comment:
This ignores strict mode. When strict mode is off, AbstractObject needs to be coerced into AbstractArray